.github/workflows/tests.yml

```
@@ -15,21 +16,21 @@ on:
     branches: master
   schedule:
     # * is a special character in YAML so you have to quote this string
-    - cron:  '0 3 * * 6'
+    - cron: "0 3 * * 6"
   workflow_dispatch:
     inputs:
       reason:
-        description: 'Reason'
+        description: "Reason"
         required: false
-        default: 'Manual trigger'
+        default: "Manual trigger"
 
 jobs:
   Tests:
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
       matrix:
-          python-version: [3.8, 3.9]
-          requirements-level: [pypi]
+        python-version: [3.9, 3.12]
```

```suggestion
        python-version: ["3.9", "3.12"]
```

nit: this was an issue for `3.10` which got interpreted as the floating point 3.1 value. I guess it's going to be eventually an issue for 3.20 😅